src/instructions/interrupts.rs

```diff
@@ -28,10 +28,11 @@ pub fn enable() {
 /// This is a wrapper around the `cli` instruction.
 #[inline]
 pub fn disable() {
-    // Prevent future writes to be moved before this point.
-    compiler_fence(Ordering::Acquire);
     unsafe {
-        asm!("cli", options(nomem, nostack));
+        // Omit `nomem` to imitate a lock acquire.
+        // Otherwise, the compiler is free to move
+        // reads and writes through this asm block.
+        asm!("cli", options(preserves_flags, nostack));
     }
```

Comment:
```suggestion
    // Omit `nomem` to imitate a lock acquire. Otherwise, the compiler
    // is free to move reads and writes through this asm block.
    unsafe {
        asm!("cli", options(preserves_flags, nostack));
    }
```